src/main/resources/com/google/api/codegen/java/main.snip

```diff
@@ -670,54 +668,61 @@
   {@context.basicTypeNameBoxed(resourcesField.getType())}
 @end
 
-@private initLine(param, docConfig)
-  @let paramType = param.getType()
-    @if paramType.isMessage()
-      @if docConfig.getRequiredParams.isEmpty
-        {@initLineSimple(param, docConfig)}
-      @else
-        {@initLineMessage(param, docConfig)}
-      @end
-    @else
-      {@initLineSimple(param, docConfig)}
+@private initCode(initCodeSpec)
+  @join line : initCodeSpec.getLines()
+    @switch line.getLineType.toString()
+    @case "StructureInitLine"
+      {@initLineStructure(line)}
+    @case "ListInitLine"
+      {@initLineList(line)}
+    @case "SimpleInitLine"
+      {@initLineSimple(line)}
+    @default
+      ??
```

Comment:
This seems like it should fail fast (i.e. calling a non-existent function) rather than silently putting ?? into the snippet.